src/color.rs

```diff
@@ -416,6 +609,121 @@ mod tests {
         assert!(!Color {red: 0.0, green: 0.0, blue: 0.0, alpha: 1.0001}.is_valid());
         assert!(!Color {red: 255.0001, green: 255.0001, blue: 255.0001, alpha: 1.0001}.is_valid());
     }
+
+    #[test]
+    fn ensure_value_mapping() {
+        let expected = Color { red: 65., green: 122., blue: 200., alpha: 1.};
+        let actual_rgb = Color::rgb(65., 122., 200.);
+        let actual_rgba = Color::rgba(65., 122., 200., 1.);
+        assert_eq!(expected, actual_rgb);
+        assert_eq!(expected, actual_rgba);
+    }
+
+    #[test]
+    fn ensure_hsl() {
+        let expected = Color { red: 148., green: 139., blue: 193., alpha: 1.};
+        let actual = Color::hsl(250., 0.3, 0.65);
+        assert_eq!(expected, actual);
+    }
+
+    #[test]
+    fn ensure_hsla() {
+        let expected = Color { red: 148., green: 139., blue: 193., alpha: 1.};
+        let actual = Color::hsla(250., 0.3, 0.65, 1.0);
+        assert_eq!(expected, actual);
+    }
+
+    #[test]
+    fn ensure_achromatic_hsl() {
+        let expected = Color { red: 26., green: 26., blue: 26., alpha: 1.};
+        let actual = Color::hsl(180., 0., 0.1);
+        assert_eq!(expected, actual);
+    }
+
+    #[test]
+    fn check_rgb_values() {
+        rgb_mapping_values().iter().for_each(|t| assert_eq!(Color::from(t.0), Color::rgb((t.1).0, (t.1).1, (t.1).2)));
```

Comment:
I think you could make these tests more readable using an [irrefutable pattern](https://doc.rust-lang.org/book/second-edition/ch18-01-all-the-places-for-patterns.html#function-parameters).

Something like this:
```rust
rgb_mapping_values().iter().for_each(|&(expected, (r, g, b))| assert_eq!(Color::from(expected), Color::rgb(r, g, b)));
```

Reply:
Sure